src/aggregates/mod.rs

```diff
@@ -0,0 +1,189 @@
+use std::collections::VecDeque;
```

Comment:
should this be called gapfill_delta.rs?

Reply:
Kinda maybe, but probably not directly. Rust's module system expects there to be a module for `aggregates` somewhere (because we have defined `mod aggregates;` in `lib.rs`. The `aggregates` module implementation can either be in `aggregates/mod.rs` (this file) or in `aggregates.rs` (which doesn't exist). This means that we can't rename `mod.rs` to `gapfill_delta.rs`. We could move the `GapfillDelta...` stuff into a separate source file named `gapfill_delta.rs`, but I'm not sure that that adds any value.

@bitwalker maybe you could add your thoughts here. I'm not a Rust guru.

Reply:
As a general rule, I use the `<module>/mod.rs` convention whenever there are multiple submodules, and use `mod.rs` to hold module-global stuff (like custom error/result types, constants, etc), and to re-export submodule types/functions/etc., that the module aims to provide; it's also a good place to define any `macro_rules!` macros used within the module. For anything else, I use the single-file `<module>.rs` convention. 

The approach @JamesGuthrie used here feels right to me. I do agree though that the `GapfillDeltaTransition` bits probably make sense to define in a submodule; the rest of it seems like it is in the right place, at least for now.

Reply:
Ok I do think we should move GapfillDeltaTransition but will leave it up to you in this PR.

Reply:
done